core/src/test/java/org/elasticsearch/index/percolator/ExtractQueryTermsServiceTests.java

```diff
@@ -245,6 +251,59 @@ public void testExtractQueryMetadata_blendedTermQuery() {
         assertThat(result.get(1).text(), equalTo("_term2"));
     }
 
+    public void testExtractQueryMetadata_spanQueries() {
+        // the following queries aren't exposed in the query dsl and are therefor not supported:
+        // 1) SpanPositionRangeQuery
+        // 2) PayloadScoreQuery
+        // 3) SpanBoostQuery
+
+        // The following span queries can't be supported because of how these queries work:
+        // 1) SpanMultiTermQueryWrapper, not supported, because there is no support for MTQ typed queries yet.
+        // 2) SpanContainingQuery, is kind of range of spans and we don't know what is between the little and big terms
+        // 3) SpanWithinQuery, same reason as SpanContainingQuery
+
+        SpanTermQuery spanTermQuery1 = new SpanTermQuery(new Term("_field", "_short_term"));
+        List<Term> terms = new ArrayList<>(ExtractQueryTermsService.extractQueryTerms(spanTermQuery1));
+        assertThat(terms.size(), equalTo(1));
+        assertThat(terms.get(0).field(), equalTo(spanTermQuery1.getTerm().field()));
+        assertThat(terms.get(0).bytes(), equalTo(spanTermQuery1.getTerm().bytes()));
+
+        SpanTermQuery spanTermQuery2 = new SpanTermQuery(new Term("_field", "_very_long_term"));
+        SpanNearQuery spanNearQuery = new SpanNearQuery.Builder("_field", true)
+                .addClause(spanTermQuery1).addClause(spanTermQuery2).build();
+        terms = new ArrayList<>(ExtractQueryTermsService.extractQueryTerms(spanNearQuery));
+        assertThat(terms.size(), equalTo(1));
+        assertThat(terms.get(0).field(), equalTo(spanTermQuery2.getTerm().field()));
+        assertThat(terms.get(0).bytes(), equalTo(spanTermQuery2.getTerm().bytes()));
+
+        SpanOrQuery spanOrQuery = new SpanOrQuery(spanTermQuery1, spanTermQuery2);
+        terms = new ArrayList<>(ExtractQueryTermsService.extractQueryTerms(spanOrQuery));
+        Collections.sort(terms);
+        assertThat(terms.size(), equalTo(2));
+        assertThat(terms.get(0).field(), equalTo(spanTermQuery1.getTerm().field()));
+        assertThat(terms.get(0).bytes(), equalTo(spanTermQuery1.getTerm().bytes()));
+        assertThat(terms.get(1).field(), equalTo(spanTermQuery2.getTerm().field()));
+        assertThat(terms.get(1).bytes(), equalTo(spanTermQuery2.getTerm().bytes()));
```

Comment:
Can we rather do something like:
```
assertEquals(new HashSet<>(spanTermQuery1.getTerm(), spanTermQuery2.getTerm()), terms);
```
to avoid having to sort?